src/app/core/device.ts

```diff
@@ -0,0 +1,1044 @@
+/// <reference types="node" />
+
+import Params from "./params";
+
+import StringValue from "./parameters/stringValue";
+import Feature from "./feature";
+import {DeviceInterchangeV1} from "./init"
+import {ComponentInterchangeV1} from "./init"
+import {ConnectionInterchangeV1} from "./init"
+import {LayerInterchangeV1} from "./init"
+import {FeatureInterchangeV1} from "./init"
+
+import Registry from './registry';
+
+import Layer from "./layer";
+import Component from "./component";
+import Connection from "./connection";
+import EdgeFeature from "./edgeFeature";
+import DXFObject from "./dxfObject";
+import * as FeatureSets from "../featureSets";
+import Valve from "../library/valve";
+import ComponentPort from "./componentPort";
+import * as IOUtils from "../utils/ioUtils";
+
+
+/**
+ * The Device stores information about a design.
+ */
+export default class Device {
+    private __layers: Array<Layer>;
+    private __textLayers: Array<Layer>;
+    private __params: Params;
+    private __name: StringValue;
+    private __components: Array<Component>;
+    private __nameMap: Map<string,number>;
+    private __connections: Array<Connection>;
+    private __valveMap: Map<string,string>;
+    private __valveIs3DMap: Map<string,boolean>;
+    private __groups: Array<string>;
+    private __features: Array<Feature>;
+    private __version: number;
+    private __defaults: ;
```

Comment:
'defaults' is only used in toJSON and fromJSON, so there is no way to trace it to other files. I am unsure what if any purpose it serves. I made a JSON device file from the website and defaults was not contained within it.